ush/forecast_predet.sh

```diff
@@ -54,7 +54,15 @@ common_predet(){
   current_cycle_begin=$(date --utc -d "${current_cycle:0:8} ${current_cycle:8:2} - ${half_window} hours" +%Y%m%d%H)
   current_cycle_end=$(date --utc -d "${current_cycle:0:8} ${current_cycle:8:2} + ${half_window} hours" +%Y%m%d%H)
   next_cycle_begin=$(date --utc -d "${next_cycle:0:8} ${next_cycle:8:2} - ${half_window} hours" +%Y%m%d%H)
-  next_cycle_end=$(date --utc -d "${next_cycle:0:8} ${next_cycle:8:2} + ${half_window} hours" +%Y%m%d%H)
+  forecast_end_cycle=$(date --utc -d "${current_cycle:0:8} ${current_cycle:8:2} + ${FHMAX} hours" +%Y%m%d%H)
+  #Define sdate for current_cycle and next_cycle as the time the forecast will start for this cycle and next cycle
+  if [[ "${DOIAU:-}" == "YES" ]]; then
+    sdate_current_cycle=$(date --utc -d "${current_cycle:0:8} ${current_cycle:8:2} - ${half_window} hours" +%Y%m%d%H)
+    sdate_next_cycle=$(date --utc -d "${next_cycle:0:8} ${next_cycle:8:2} - ${half_window} hours" +%Y%m%d%H)
```

Comment:
These are calculated on the few lines above and it makes casts `model_start_date_current_cycle` and `model_start_date_next_cycle` in relation to the anchors above.
```suggestion
    sdate_current_cycle="${current_cycle_begin}"
    sdate_next_cycle="${next_cycle_begin}"
```